src/bridge/test-httpstream.c

```diff
@@ -72,6 +72,7 @@ setup_general (TestGeneral *tt,
                gconstpointer host_fixture)
 {
   tt->web_server = cockpit_web_server_new (NULL, 0, NULL, COCKPIT_WEB_SERVER_NONE, NULL, NULL);
+  cockpit_web_server_set_redirect_tls (tt->web_server, FALSE);
```

Comment:
What is the bug that causes this to be currently off?

Reply:
The one fixed in the later commit:
```diff
--- a/src/common/test-webserver.c
+++ b/src/common/test-webserver.c
@@ -1245,7 +1264,7 @@ cockpit_request_start (CockpitWebServer *self,
       socket = g_socket_connection_get_socket (connection);
       g_socket_set_blocking (socket, FALSE);
 
-      if (self->certificate)
+      if (self->certificate || self->redirect_tls)
         {
           request->source = g_socket_create_source (g_socket_connection_get_socket (connection),
                                                     G_IO_IN, NULL);

```